src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -466,7 +466,10 @@ private static Continuation GetHeadContinuation(AwaitableProxy awaitableProxy)
         {
             ref RuntimeAsyncAwaitState state = ref t_runtimeAsyncAwaitState;
             awaitableProxy._notifier = state.Notifier;
-            return state.SentinelContinuation!.Next!;
+            Continuation sentinelContinuation = state.SentinelContinuation!;
+            Continuation head = sentinelContinuation.Next!;
+            sentinelContinuation.Next = null;
```

Comment:
Perhaps rename the function to something like `UnlinkHeadContinuation` - to imply this cannot be done twice.

Reply:
Good idea, renamed!